File: core/core-enum/src/main/java/com/imstargg/core/enums/TrophyRange.java
package com.imstargg.core.enums;

public enum TrophyRange {

    TROPHY_0_PLUS(0),
    TROPHY_500_PLUS(500),
    TROPHY_1000_PLUS(1000),
    ;

    private final int minTrophy;

    TrophyRange(int minTrophy) {
        this.minTrophy = minTrophy;
    }

    public int getMinTrophy() {
        return minTrophy;
    }
}

File: imstargg-frontend/src/model/enums/TrophyRange.ts
export const TrophyRangeValue = {
  TROPHY_0_PLUS: 'TROPHY_0_PLUS',
  TROPHY_500_PLUS: 'TROPHY_500_PLUS',
  TROPHY_1000_PLUS: 'TROPHY_1000_PLUS',
} as const;

export type TrophyRange = typeof TrophyRangeValue[keyof typeof TrophyRangeValue];

export const TrophyRangeValues = Object.keys(TrophyRangeValue) as TrophyRange[];

export const trophyRangeTitle = (trophy: TrophyRange): string => {
  switch (trophy) {
    case TrophyRangeValue.TROPHY_0_PLUS:
      return '0+';
    case TrophyRangeValue.TROPHY_500_PLUS:
      return '500+';
    case TrophyRangeValue.TROPHY_1000_PLUS:
      return '1000+';
  }
}
